pkg/sql/opt/memo/testdata/logprops/scan

```diff
@@ -175,3 +175,32 @@ select
            └── lt [type=bool]
                 ├── variable: d [type=date]
                 └── const: '2017-01-05' [type=date]
+
+
+# Regression test for #42731: we were incorrectly setting cardinality [0 - 1].
+exec-ddl
+CREATE TABLE t42731 (id INT PRIMARY KEY, unique_value INT UNIQUE, notnull_value INT NOT NULL)
+----
+
+norm
+SELECT * FROM t42731 WHERE unique_value IS NULL AND notnull_value = 2000
+----
+select
+ ├── columns: id:1(int!null) unique_value:2(int) notnull_value:3(int!null)
+ ├── key: (1)
+ ├── fd: ()-->(2,3), (2)~~>(1)
+ ├── prune: (1)
+ ├── interesting orderings: (+1) (+2,+1)
+ ├── scan t42731
+ │    ├── columns: id:1(int!null) unique_value:2(int) notnull_value:3(int!null)
+ │    ├── key: (1)
+ │    ├── fd: (1)-->(2,3), (2)~~>(1,3)
```

Comment:
I'm trying to backport to 19.1, and there we don't have the `(2) ~~> (1,3)` relation which seems very strange. Does anyone remember fixing something in this area? @rytaft @justinj

Reply:
Ah, never mind. The test catalog in 19.1 just doesn't understand the "inline" `UNIQUE`